model-extension-example/src/main/java/uk/gov/gchq/magmacore/examples/extensions/ext/HqdmExtensionService.java

```diff
@@ -16,21 +19,29 @@ public class HqdmExtensionService implements ExtensionService {
     /**
      * {@inheritDoc}
      */
+    @SuppressWarnings("unchecked")
     @Override
-    public Thing createEntity(final String typeName, final IRI iri) {
-
-        // Check the type name and return an appropriate instance.
-        if (Constants.UK_LIMITED_COMPANY_TYPE_NAME.equals(typeName)) {
-            final var result = new UkLimitedCompanyImpl(iri);
-            result.addValue(RDFS.RDF_TYPE, Constants.UK_LIMITED_COMPANY_IRI);
-            return result;
-        } else if (Constants.UK_SOFTWARE_DEVELOPMENT_COMPANY_TYPE_NAME.equals(typeName)) {
-            final var result = new UkSoftwareDevelopmentCompanyImpl(iri);
-            result.addValue(RDFS.RDF_TYPE, Constants.UK_SOFTWARE_DEVELOPMENT_COMPANY_IRI);
-            return result;
-        }
+    public <T extends Thing> T createEntity(final String typeName, final IRI iri) {
+
+        switch (typeName) {
+            case UK_LIMITED_COMPANY_TYPE_NAME:
+                return (T) new UkLimitedCompanyImpl(iri);
+
+            case UK_SOFTWARE_DEVELOPMENT_COMPANY_TYPE_NAME:
+                return (T) new UkSoftwareDevelopmentCompanyImpl(iri);
 
-        // Or null if it's not a class from this extension package.
-        return null;
+            case PARENT_TYPE_NAME:
+                return (T) new ParentImpl(iri);
+
+            case CHILD_TYPE_NAME:
+                return (T) new ChildImpl(iri);
+
+            case PARENT_CHILD_ASSOCIATION_TYPE_NAME:
+                return (T) new ParentChildAssociationImpl(iri);
+
+            default:
+                // Or null if it's not a class from this extension package.
+                return null;
```

Comment:
Would it be preferable to throw an exception?

Reply:
I did consider that but decide that it's normal for an extension JAR not to be able to handle the request, so it's not an exceptional situation which means `null` should be ok. Another alternative would be to return an empty  `Optional<T>` instead rather than relying on `null`, which I can do if you prefer.

Reply:
What do you think is best from a consumer perspective?

Reply:
The code looks neater without having to handle the `Optional` value and it's mostly internal, so in this case returning `null` is _slightly_ better, but it's marginal I think.